2025-10-21 14:27:50,459 [processing](INFO): Initializing backseat process (pilot lrauv based on marl)
2025-10-21 14:27:54,187 [Logger](CRITICAL): Uncaught exception
Traceback (most recent call last):
  File "./backseat_app/main.py", line 111, in <module>
    main()
  File "./backseat_app/main.py", line 107, in main
    backseat_app.spin()
  File "./backseat_app/main.py", line 75, in spin
    self.marl_processor.compute_new_heading()
  File "/home/maiv/lrauv_sim/lrauv-backseat-marl/backseat_app/processing.py", line 200, in compute_new_heading
    self.new_action = self.rl_tracking.newAction(self.target_address,agents_range,agents_pose,agents_depth,agents_timestamp,new_range=aux_range)
  File "/home/maiv/lrauv_sim/lrauv-backseat-marl/backseat_app/utils.py", line 992, in newAction
    self.action, self.target_predictions = self.agent_controller.get_action_and_predictions(
  File "/home/maiv/lrauv_sim/lrauv-backseat-marl/backseat_app/../jaxagent/jaxagent/production_agent_torch.py", line 110, in get_action_and_predictions
    action = self.actor.step(obs, done=False)
  File "/home/maiv/lrauv_sim/lrauv-backseat-marl/backseat_app/../jaxagent/jaxagent/torch_agent.py", line 498, in step
    self.debug_raw_observation(obs['agent_0'])
  File "/home/maiv/lrauv_sim/lrauv-backseat-marl/backseat_app/../jaxagent/jaxagent/torch_agent.py", line 413, in debug_raw_observation
    print(f"  Observation keys: {list(agent_obs.keys())}")
AttributeError: 'numpy.float64' object has no attribute 'keys'
